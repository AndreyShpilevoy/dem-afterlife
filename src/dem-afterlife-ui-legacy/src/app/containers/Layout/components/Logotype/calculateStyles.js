const calculateStyles = ({logotype, focus}) => ({
    logotype: {
        backgroundImage: `url(${logotype.backgroundImage})`,
        backgroundRepeat: logotype.backgroundRepeat,
        backgroundSize: logotype.backgroundSize,
        height: '100%'
    },
    logotypeContainer: {
        height: logotype.height,
        '&:focus': {
            outlineColor: focus.colorLight
        }
    }
});

export default calculateStyles;