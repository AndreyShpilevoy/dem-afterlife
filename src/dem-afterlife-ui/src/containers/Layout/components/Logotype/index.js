import React from "react";
import { string, shape } from "prop-types";
import Link from "components/Link";
import { injectSheet } from "styles";
import calculateStyles from "./calculateStyles";

export const LogotypePure = ({ classes, className }) => {
  const linkClassName = `${classes.logotypeContainer} ${className || ""}`;
  return (
    <Link className={linkClassName} to="/">
      <div className={classes.logotype} />
    </Link>
  );
};

LogotypePure.displayName = "LogotypePure";

LogotypePure.propTypes = {
  classes: shape().isRequired,
  className: string
};

LogotypePure.defaultProps = {
  className: ""
};

export default injectSheet(calculateStyles, { componentName: "Logotype" })(
  LogotypePure
);
