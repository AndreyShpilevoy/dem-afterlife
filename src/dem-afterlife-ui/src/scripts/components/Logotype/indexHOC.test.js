/* eslint no-undef: 0, fp/no-unused-expression: 0, fp/no-nil: 0*/

import React from 'react';
import {shallow} from 'enzyme';
import Logotype from './index';

describe('Logotype HOC', () => {
    it('component match expected snapshot', () => {
        expect(shallow(<Logotype theme={'default'} />)).toMatchSnapshot();
    });
});
