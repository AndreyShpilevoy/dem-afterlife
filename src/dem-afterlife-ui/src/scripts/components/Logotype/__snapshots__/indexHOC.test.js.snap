// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logotype HOC component match expected snapshot 1`] = `
<LogotypePure
  styles={
    Object {
      "logotype": "default-Logotype-logotype",
      "logotypeContainer": "default-Logotype-logotypeContainer",
    }
  }
/>
`;
